src/planner/binder/statement/bind_create.cpp

```diff
@@ -630,22 +630,28 @@ BoundStatement Binder::Bind(CreateStatement &stmt) {
 	case CatalogType::DATABASE_ENTRY: {
 		// not supported in DuckDB yet but allow extensions to intercept and implement this functionality
 		auto &base = (CreateDatabaseInfo &)*stmt.info;
-		string extension_name = base.extension_name;
 		string database_name = base.name;
 		string source_path = base.path;
 
 		auto &config = DBConfig::GetConfig(context);
-		for (auto &extension : config.create_database_extensions) {
-			auto create_database_function_ref =
-			    extension.function(context, extension_name, database_name, source_path, extension.data.get());
-			if (create_database_function_ref) {
-				auto bound_create_database_func = Bind(*create_database_function_ref);
-				result.plan = CreatePlan(*bound_create_database_func);
-				break;
-			}
-		}
-		if (!result.plan) {
+
+		// for now only handling the case where there is one storage extension registered
+		if (config.storage_extensions.empty()) {
 			throw NotImplementedException("CREATE DATABASE not supported in DuckDB yet");
+		} else if (config.storage_extensions.size() > 1) {
+			throw BinderException("CREATE DATABASE only supported when there is one storage extension registered.");
```

Comment:
Perhaps you can instead loop over the storage extensions and find one that has `create_database` set to a non-null pointer? Otherwise you cannot use the SQLite extension together with the Motherduck extension.

Reply:
Done